### ISSUE TITLE
Timezone in `parse()` is not converted to string

### ISSUE THREAD

**Issue by stebogit** [briannesbitt/Carbon#1958]
Hello,
I encountered an issue with the following PHPUnit code:
```php
$date = '2018-02-01';
echo Carbon::parse($date, -8)->toIso8601String();
echo "\n";
echo Carbon::parse($date, '-8')->toIso8601String();
```

Carbon version: **2.27.0**
PHP version: **7.3.4-1+ubuntu18.04.1+deb.sury.org+3** (from Homestead v7.2.1)

I expected to get:

```
2018-02-01T00:00:00-08:00
2018-02-01T00:00:00-08:00
```

But I actually get:

```
2018-02-01T00:00:00-01:00
2018-02-01T00:00:00-08:00
```

I might be wrong, but I would expect that the timezone would be interpreted as a string also in the constructor, like in `setTimezone()` where the type of the input does not matter:

```php
Carbon::parse($date)->setTimezone(-8)->toIso8601String();   // 2018-01-31T16:00:00-08:00
Carbon::parse($date)->setTimezone('-8')->toIso8601String(); // 2018-01-31T16:00:00-08:00
```

**Comment by kylekatarnls** [comment]
It's a legacy stuff kept for backward compatibility. But I think we can get rid of it. I have to check the unit test that a fix for this would break.

You still can get no ambiguity using:
```php
echo Carbon::parse($date, CarbonTimeZone::create(-8))->toIso8601String();
```

**Comment by kylekatarnls** [comment]
Here is the unit test that show the fix:
https://github.com/briannesbitt/Carbon/pull/1959/files#diff-e647fc4a40b22cf61c3f64c24341ffaeL118

So next version will no longer try to auto-detect time zone name from the offset and will just use the offset.

Let me give some advice about timezones, you should work as long as you can with UTC, it's the **universal** way to represent time, while -8 is an arbitrary timezone offset that can be used by 0, 1 or many regions, and they can use it differently (with or without DST), and region can decide to change their timezone. So use UTC for create and manipulate server side dates, then change the timezone using `tz()` just before to display/send the date and prefer to use named timezone (America/New_York or Europe/London, etc.), or the best, send UTC ISO string and let the client handle the conversion.

I detailed the subject deeper here:
https://medium.com/@kylekatarnls/handle-dates-the-right-way-in-php-c215650fc4fa
https://medium.com/@kylekatarnls/always-use-utc-dates-and-times-8a8200ca3164

**Comment by kylekatarnls** [comment]
Will be fixed in 2.28.0
